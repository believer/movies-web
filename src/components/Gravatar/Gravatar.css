/* ==================================================================
   Gravatar
   ================================================================== */

.Gravatar {
  border-radius: 100%;
  flex: 1;
  max-width: 40px;
  overflow: hidden;
  position: relative;
  transition: all 200ms ease-in-out;
  z-index: 1;
}

.Gravatar:hover {
  border-color: var(--color-radical-red);
}

.Gravatar__image {
  display: block;
  max-width: 100%;
}

.small {
  border: 0;
  max-width: 30px;
}
