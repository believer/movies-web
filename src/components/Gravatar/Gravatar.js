// @flow

import './Gravatar.css'
import React from 'react'
import md5 from 'md5'
import classnames from 'classnames'

type Props = {
  alt?: string,
  className?: string,
  email: string,
  size: number
}

const Gravatar = ({ className, email, alt, size }: Props) => {
  if (!email) {
    return null
  }

  let gravatar = `https://www.gravatar.com/avatar/${md5(email)}`

  if (size) {
    gravatar = `${gravatar}&s=${size}`
  }

  return (
    <div
      className={classnames('Gravatar', className)}
      style={{ maxWidth: size, width: size, height: size }}
    >
      <img alt={alt} className="Gravatar__image" src={gravatar} />
    </div>
  )
}

Gravatar.defaultProps = {
  email: '',
}

export default Gravatar
