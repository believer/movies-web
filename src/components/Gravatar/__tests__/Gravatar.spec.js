import React from 'react'
import { shallow } from 'enzyme'
import Gravatar from '../Gravatar'

jest.mock('md5', () => jest.fn(() => 'md5'))

describe('components/Gravatar', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <Gravatar alt="Name" email="cookie@monster.com" size={50} />
    )
  })

  it('renders Gravatar', () => {
    expect(component).toMatchSnapshot()
  })
})
