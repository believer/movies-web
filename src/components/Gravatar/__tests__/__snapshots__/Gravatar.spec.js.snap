// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Gravatar renders Gravatar 1`] = `
<div
  className="Gravatar"
  style={
    Object {
      "height": 50,
      "maxWidth": 50,
      "width": 50,
    }
  }
>
  <img
    alt="Name"
    className="Gravatar__image"
    src="https://www.gravatar.com/avatar/md5&s=50"
  />
</div>
`;
